sips/sip-xyz/sip-xyz-bitwise-ops.md

```diff
@@ -0,0 +1,203 @@
+(^ -128 64) ;; Returns -64
+(^ u24 u4) ;; Returns u28
+(^ 1 2 4 -1) ;; Returns -8
+```
+
+## Bitwise And (`&`)
+
+`(& i1 i2...)`
+
+- **Inputs:** int, ... | uint, ...
+- **Output:** int | uint
+
+Returns the result of bitwise and'ing a variable number of integer inputs.
+
+### Examples
+
+```
+(& 24 16) ;; Returns 16
+(& 28 24 -1) ;; Returns 24
+(& u24 u16) ;; Returns u16
+(& -128 -64) ;; Returns -128
+(& 28 24 -1) ;; Returns 24
+```
+
+## Bitwise Or (`|`)
+
+`(& i1 i2...)`
+
+- **Inputs:** int, ... | uint, ...
+- **Outputs:** int | uint
+
+Returns the result of bitwise inclusive or'ing a variable number of integer
+inputs.
+
+### Examples
+
+```
+(| 4 8) ;; Returns 12
+(| 1 2 4) ;; Returns 7
+(| 64 -32 -16) ;; Returns -16
+(| u2 u4 u32) ;; Returns u38
+```
+
+## Bitwise Not (`~`)
+
+`(~ i1)`
+
+- **Inputs:** int | uint
+- **Output:** int | uint
+
+Returns the one's compliment (sometimes also called the bitwise compliment or
+not operator) of `i1`, effectively reversing the bits in `i1`.
+
+In other words, every bit that is `1` in `ì1` will be `0` in the result.
+Conversely, every bit that is `0` in `i1` will be `1` in the result.
+
+### Examples
+
+```
+(~ 3) ;; Returns -4
+(~ u128) ;; Returns u340282366920938463463374607431768211327
+(~ 128) ;; Returns -129
+(~ -128) ;; Returns 127
+```
+
+## Bitwise Left Shift (`<<`)
+
+`(<< i1 shamt)`
+
+- **Inputs:** int, uint | uint, uint
+- **Outputs:** int | uint
+
+Shifts all bits in `i1` to the left by the number of places specified in `shamt`
+modulo 128 (the bit width of Clarity integers). New bits are filled with zeros.
+
+### Examples
+
+```
+(<< 2 u1) ;; Returns 4
+(<< 16 u2) ;; Returns 64
+(<< -64 u1) ;; Returns -128
+(<< u4 u2) ;; Returns u16
+(<< u240282366920938463463374607431768211327 u3) ;; Returns 30756142965880123323311949751266331049856
+(<< u123 u24028236699) ;; Returns 16508780544 (123 << 27)
+```
+
+## Bitwise Right Shift (`>>`)
+
+`(>> i1 shamt)`
+
+- **Inputs:** int, uint | uint, uint
+- **Output:** int | uint
+
+Shifts all the bits in `i1` to the right by the number of places specified in
+`shamt` modulo 128 (the bit width of Clarity integers). When `i1` is a `uint`
+(unsigned), new bits are filled with zeros. When `i1` is an `int` (signed), the
+sign is preserved, meaning that new bits are filled with the value of the
+previous sign-bit.
+
+### Examples
+
+```
+(>> 2 u1) ;; Returns 1
+(>> 128 u2) ;; Returns 32
+(>> -64 u1) ;; Returns -32
+(>> u128 u2) ;; Returns u32
+(>> u240282366920938463463374607431768211327 u127) ;; Returns 0
+(>> u123 u2402820) ;; Returns 7 (u123 >> 4)
+(>> -3 u12) ;; Returns -1
+```
+
+# Related work
+
+Not applicable
+
+# Backwards Compatibility
+
+Because this SIP introduces new Clarity operators, it is a consensus-breaking
+change. A contract that uses one of these new operators would be invalid before
+this SIP is activated, and valid after it is activated.
+
+# Activation
+
+This SIP will be a rider on SIP-015. It will be considered activated if SIP-015
```

Comment:
Minor nit -- "It will be considered activated if **and only if** SIP-015 is activated".  This SIP should not active if SIP 015 fails to activate.

Reply:
Added in 166ae07.